File: API_RestaurantManagement/src/main/java/com/example/API_RestaurantManagement/entity/NhanVienEntity.java
package com.example.API_RestaurantManagement.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.Size;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "NHANVIEN")
public class NhanVienEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "IDNV")
    private Integer idNV;

    @Column(name = "HOTEN")
    @Size(max = 50)
    private String hoTen;

    @Column(name = "SDT")
    @Size(max = 50)
    private String sDT;

    @Column(name = "DIACHI")
    @Size(max = 50)
    private String diaChi;

    @Column(name = "EMAIL")
    @Size(max = 50)
    private String email;

    @OneToMany(cascade = CascadeType.REFRESH, mappedBy = "idnv")
    private List<PhieuNhapNguyenLieuEntity> phieunhapnguyenlieuList;

    @OneToOne(cascade = CascadeType.REFRESH, mappedBy = "idnv")
    private TaiKhoanEntity taikhoan;

    @OneToMany(cascade = CascadeType.REFRESH, mappedBy = "idnv")
    private List<PhieuDatEntity> phieudatList;

    @OneToMany(cascade = CascadeType.REFRESH, mappedBy = "idnv")
    private List<ThayDoiGiaMonEntity> thaydoigiamonList;

    @ManyToOne
    @JoinColumn(name = "MAPB")
    private BoPhanEntity mapb;

    @OneToMany(cascade = CascadeType.REFRESH, mappedBy = "idnv")
    private List<HoaDonEntity> hoadonList;
}

File: OrderManagement/app/src/main/java/com/example/ordermanagement/model/NhanVienEntity.kt
package com.example.ordermanagement.model

import com.squareup.moshi.Json

data class NhanVienEntity(
    @Json(name = "idNV") val idNV: Int,
    @Json(name = "hoTen") val hoTen: String,
    @Json(name = "diaChi") val diaChi: String,
    @Json(name = "email") val email: String,
    @Json(name = "taiKhoan") val taiKhoan: String,
    @Json(name = "maBP") val maBP: String,
    @Json(name = "tenBP") val tenBP: String,
    @Json(name = "sdt") val sdt: String
)
